xapian-letor/tests/api_letor.cc

```diff
@@ -104,26 +98,19 @@ gen_uniqterms_gt_doclen_db(Xapian::WritableDatabase& db, const string&)
     db.add_document(doc);
 }
 
-DEFINE_TESTCASE(ranker1, generated) {
+DEFINE_TESTCASE(listnet_ranker, generated) {
     Xapian::ListNETRanker ranker;
     string a = get_database_path("apitest_ranker1",
-				       gen_uniqterms_gt_doclen_db);
+				       populate_db);
     Xapian::Enquire enquire((Xapian::Database(a)));
     enquire.set_query(Xapian::Query("lions"));
     Xapian::MSet mymset = enquire.get_mset(0, 10);
-    string query ="20001 'lions'";
-    string qrel = "20001 Q0 1 1\n20001 Q0 2 2";
-    std::ofstream myfile;
-    myfile.open("query.txt");
-    myfile << query;
-    myfile.close();
-    myfile.open("qrel.txt");
-    myfile << qrel;
-    myfile.close();
-    Xapian::prepare_training_file(a, "query.txt", "qrel.txt", 10);
+    Xapian::prepare_training_file(a, "testdata/query.txt",
+				  "testdata/qrel.txt", 10,
```

Comment:
You need to support srcdir != builddir (also known as a "VPATH build", but when that's the case, the current directory will be the builddir and `testdata` will be in the srcdir, so these two files won't be found with these relative paths.

You can get the appropriate directory with `test_driver::get_srcdir()` - look at the code which loads files from `testdata` in the `xapian-core/tests` subdirectory.